pandas/core/window_/indexers.py

```diff
@@ -0,0 +1,226 @@
+            A tuple of ndarray[int64]s, indicating the boundaries of each
+            window
+
+        """
+
+
+class FixedWindowIndexer(BaseIndexer):
+    """Calculate window boundaries that have a fixed window size"""
+    def get_window_bounds(
+        self,
+        values: Optional[np.ndarray] = None,
+        window_size: int = 0,
+        min_periods: Optional[int] = None,
+        center: Optional[bool] = None,
+        closed: Optional[str] = None,
+        win_type: Optional[str] = None,
+    ):
+        """
+        Examples
+        --------
+        >>> FixedWindowIndexer().get_window_bounds(np.arange(10), 2)
+        (array([0, 0, 1, 2, 3, 4, 5, 6, 7, 8]),
+         array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10]))
+
+        >>> FixedWindowIndexer().get_window_bounds(np.arange(10), 2)
+        (array([0, 0, 0, 1, 2]), array([1, 2, 3, 4, 5]))
+        """
+        num_values = len(values) if values is not None else 0
+        start_s = np.zeros(window_size, dtype=np.int64)
+        start_e = np.arange(window_size, num_values, dtype=np.int64) - window_size + 1
+        start = np.concatenate([start_s, start_e])
+
+        end_s = np.arange(window_size, dtype=np.int64) + 1
+        end_e = start_e + window_size
+        end = np.concatenate([end_s, end_e])
+        return start, end
+
+
+class VariableWindowIndexer(BaseIndexer):
+    """Calculate window boundaries with variable closed boundaries and index dependent"""
+    def _calculate_closed_bounds(self, closed: Optional[str]) -> Tuple[bool, bool]:
+        """
+        Evaluate the left and right closed boundary behavior based on
+        the closed parameter.
+
+        Parameters
+        ----------
+        closed : str
+            One of right, left, both, or None
+
+
+        Examples
+        --------
+        >>> VariableWindowIndexer(np.arange(10))._calculate_closed_bounds(None)
+        # Right default
+        (False, True)
+        >>> VariableWindowIndexer()._calculate_closed_bounds(None)
+        # Both default
+        (True, True)
+        >>> VariableWindowIndexer()._calculate_closed_bounds('left')
+        (True, False)
+        """
+        left_closed = False
+        right_closed = False
+
+        # if windows is variable, default is 'right', otherwise default is 'both'
+        if closed is None:
+            closed = "right" if self.index is not None else "both"
+
+        if closed == "both":
+            left_closed = True
+            right_closed = True
+
+        elif closed == "right":
+            right_closed = True
+
+        elif closed == "left":
+            left_closed = True
+
+        return left_closed, right_closed
+
+    def get_window_bounds(
+        self,
+        values: Optional[np.ndarray] = None,
+        window_size: int = 0,
+        min_periods: Optional[int] = None,
+        center: Optional[bool] = None,
+        closed: Optional[str] = None,
+        win_type: Optional[str] = None,
+    ):
+        """
+        Examples
```

Comment:
at some point if you can add Paramterse / Returns (alternatly use a shared_doc). This can be done later, if you'd make a note.